### PR TITLE
Personal Search - Selection and local storage of user product preferences, #4119

### PR DESCRIPTION
**Pull request by stephanegigandet**
New APIs to retrieve possible values for preferences and all supported product attributes
- https://fr.openfoodfacts.dev/api/v0/preferences
- https://fr.openfoodfacts.dev/api/v0/attribute_groups

New user interface to set preferences that are saved using localstorage in the browser:
- https://world.openfoodfacts.dev/product-preferences

One drawback of localstorage is that it is subdomain specific (but that should not be an issue for most users).

This is just a start, hopefully someone can make a nicer looking UI etc.

image

### CODE REVIEW

**Review by teolemon — APPROVED** [review]
Looks good to me

File: po/common/common.pot [hunk 1]
```diff
@@ -4688,10 +4688,6 @@ msgctxt "query_filter"
 msgid "Query filter"
 msgstr "Query filter"
 
-msgctxt "nova_group_producer"
```
Comment (teolemon) [inline po/common/common.pot:4691]: @stephanegigandet you are deleting this !

Comment (stephanegigandet) [reply po/common/common.pot:4691]: good catch, I have too many open branches

**Comment by stephanegigandet** [comment]
Any idea to solve this?

Check failure on line 35 in html/js/product-preferences.js

GitHub Actions
/ lint

html/js/product-preferences.js#L35
'show_user_product_preferences' is defined but never used

**Comment by svensven** [comment]
> Any idea to solve this?
> 
> Check failure on line 35 in html/js/product-preferences.js
> 
> GitHub Actions
> / lint
> 
> html/js/product-preferences.js#L35
> 'show_user_product_preferences' is defined but never used

would this work on functions as well as variables? https://eslint.org/docs/user-guide/configuring#specifying-globals

**Comment by svensven** [comment]
> 
> 
> > Any idea to solve this?
> > Check failure on line 35 in html/js/product-preferences.js
> > GitHub Actions
> > / lint
> > html/js/product-preferences.js#L35
> > 'show_user_product_preferences' is defined but never used
> 
> would this work on functions as well as variables? https://eslint.org/docs/user-guide/configuring#specifying-globals

ah, no, it doesn't. ```/* exported show_user_product_preferences */``` is what you need.
https://eslint.org/docs/rules/no-unused-vars#exported

**Comment by stephanegigandet** [comment]
> ah, no, it doesn't. `/* exported show_user_product_preferences */` is what you need.

that worked, thank you!

**Pull request merged**